### ISSUE TITLE
"Undefined method `map' for nil:NilClass" error 

### ISSUE THREAD

**Issue by TakaakiFuruse** [teampoltergeist/poltergeist#683]
(I am posting this by suggestion of @twalpole. Originally posted here https://github.com/teampoltergeist/poltergeist/issues/653#issuecomment-152789754)

I am having the same error when I visit specific website.

The error is
gems/poltergeist-1.7.0/lib/capybara/poltergeist/browser.rb:63:in `find': undefined method `map' for nil:NilClass (NoMethodError)

What I want to do is automatically login a website.
For example, If I do this.

```ruby
require "capybara"
require "capybara/dsl"
require 'capybara/poltergeist'
include Capybara::DSL

class Scraper

  include Capybara::DSL
  attr_accessor :click_sec, :goouk

  def initialize
    @click_sec = 'https://sec-sso.click-sec.com/loginweb/'
    @goouk = 'https://www.google.co.uk'
    Capybara.current_driver = :poltergeist
    Capybara.javascript_driver = :poltergeist
    Capybara.register_driver :poltergeist do |app|
      Capybara::Poltergeist::Driver.new(app, {timeout: 120, js_errors: false, phantomjs_options: ['--load-images=no', '--ignore-ssl-errors=yes'], :debug => true})
    end
  end

  def to_click_sec
    visit(click_sec)
    p has_button? "Login"
  end

  def to_google
    visit(goouk)
    p has_link? "Gmail"
  end
end

Scraper.new.to_google # => this works fine
Scraper.new.to_click_sec # => this is not
```
Debug message for "Scraper.new.to_click_sec" is here.

```
{"id":"9d99eeb4-56ee-4ea4-b234-a59e1394d960","name":"set_js_errors","args":[false]}
{"command_id":"9d99eeb4-56ee-4ea4-b234-a59e1394d960","response":true}
{"id":"379f2e1a-5895-4d2b-9c3b-39f4318b9953","name":"set_debug","args":[true]}
{"command_id":"379f2e1a-5895-4d2b-9c3b-39f4318b9953","response":true}
{"id":"391580ad-2d9f-4613-9454-c12a65e9fefc","name":"visit","args":["https://sec-sso.click-sec.com/loginweb/"]}
{"command_id":"391580ad-2d9f-4613-9454-c12a65e9fefc","response":{"status":"success"}}
{"id":"947f0f1b-8188-47a5-99fd-668edd4a974d","name":"find","args":["xpath",".//input[./@type = 'submit' or ./@type = 'reset' or ./@type = 'image' or ./@type = 'button'][((./@id = 'Login' or contains(./@value, 'Login')) or contains(./@title, 'Login'))] | .//input[./@type = 'image'][contains(./@alt, 'Login')] | .//button[(((./@id = 'Login' or contains(./@value, 'Login')) or contains(normalize-space(string(.)), 'Login')) or contains(./@title, 'Login'))] | .//input[./@type = 'image'][contains(./@alt, 'Login')]"]}
{"command_id":"947f0f1b-8188-47a5-99fd-668edd4a974d","response":{"page_id":1}}
/home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/poltergeist-1.7.0/lib/capybara/poltergeist/browser.rb:63:in `find': undefined method `map' for nil:NilClass (NoMethodError)
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/poltergeist-1.7.0/lib/capybara/poltergeist/driver.rb:120:in `find'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/poltergeist-1.7.0/lib/capybara/poltergeist/driver.rb:124:in `find_xpath'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/node/base.rb:107:in `find_xpath'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/query.rb:110:in `block in resolve_for'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/node/base.rb:80:in `synchronize'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/query.rb:106:in `resolve_for'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/node/matchers.rb:94:in `block in assert_selector'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/node/base.rb:84:in `synchronize'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/node/matchers.rb:93:in `assert_selector'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/node/matchers.rb:39:in `has_selector?'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/node/matchers.rb:251:in `has_button?'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/session.rb:698:in `block (2 levels) in <class:Session>'
        from /home/orz/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/capybara-2.5.0/lib/capybara/dsl.rb:51:in `block (2 levels) in <module:DSL>'
        from google.rb:23:in `to_click_sec'
        from google.rb:34:in `<main>'
```
phamtomjs is 1.9.8 and poltergeist is from master brunch.

**Comment by twalpole** [comment]
I've taken a quick look at this issue - and it's caused by the fact the site uses Mootools - which appears to replace the native JSON object/methods with one that only provides encode and decode methods (no parse method) -- this means that the JSON.parse method call in the string produced by Poltergeist.WebPage#stringifyCall fails when evaluated in the pages context - not sure what to do about this other than maybe adding our own parse routine to PoltergeistAgent -- @route any suggestions?

**Comment by twalpole** [comment]
maybe we can just copy the native JSON object into poltergeist at injection time

**Comment by twalpole** [comment]
This is an issue that was fixed over 5 years ago in mootools - https://github.com/mootools/mootools-core/commit/e2d71f29230cab25516c64490f788314b250f379. - seems crazy that any current site is still using it

**Comment by twalpole** [comment]
Should be fixed by PR #684 - please try the master branch

**Comment by TakaakiFuruse** [comment]
YES!! It's working!! Thank you so much!! :thumbsup::thumbsup:
